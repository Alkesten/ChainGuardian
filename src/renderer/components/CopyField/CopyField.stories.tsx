import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, text, boolean, number, select } from '@storybook/addon-knobs';
import { CopyField, MnemonicCopyField } from './CopyField';
import {useState} from "react";

storiesOf('CopyField', module).add('CopyField', () => {
    const textValue = text("value", "Test string for copy");
    const [clickedValue, setClickedValue] = useState(false);

    return (
        <>
            <CopyField 
                onCopy={()=>{setClickedValue(true)}}
                clicked={clickedValue} value={textValue} />
        </>
    )
    
}).addDecorator(withKnobs);

storiesOf('CopyField', module).add('MnemonicCopyField', () => {
    const [clickedValue, setClickedValue] = useState(false);
    const textValue = text("mnemonic value", "hold solve hurdle seed paper rely fog burden potato portion column brisk");

    return (
        <>
            <MnemonicCopyField
                onCopy={()=>{setClickedValue(true)}}
                clicked={clickedValue} value={textValue} />
        </>
    )
}).addDecorator(withKnobs);