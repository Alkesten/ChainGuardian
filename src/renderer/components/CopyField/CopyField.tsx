import * as React from "react";
import {CopyButton} from "../Button/ButtonAction";
import {clipboard} from "electron";

export interface ICopyFieldProps {
    value: string;
    label?: string;
    onCopy?: () => void;
}

export const CopyField: React.FunctionComponent<ICopyFieldProps> = (
    props: ICopyFieldProps) => {
    
    let copyToClipboard = (): void => {clipboard.writeText(props.value);};
    if(props.onCopy){
        copyToClipboard = props.onCopy;
    }

    return(
        <>
            <h3 className="copy-field-label">{props.label}</h3>
            <div className="copy-field">
                <CopyButton onClick={copyToClipboard}/>
                <div className="copy-field-body">{props.value}</div>
            </div>
        </>
    );
};


export const MnemonicCopyField: React.FunctionComponent<ICopyFieldProps> = (
    props: ICopyFieldProps) => {
    return(
        <div className="copy-field mnemonic">
            {props.value}
            <CopyButton onClick={props.onCopy}/>
        </div>
    );
};