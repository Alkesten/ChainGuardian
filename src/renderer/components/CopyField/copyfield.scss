.copy-field{
    background-color: $neutral-600;
    width: 608px;
    min-height: 60px;
    box-shadow: inset 0px 2px 6px rgba(10, 39, 51, 0.2);
    border-radius: 4px;
    color: $neutral-100;
    padding: 16px;
    font-family: 'Palanquin', sans-serif;
    font-weight: 500;
    font-size: 24px;
    line-height: 38px;
    text-align: left;
    letter-spacing: 0.01em;
    margin: 0px;
    position: relative;
    z-index: 1;
    max-height: 90px;
    overflow: auto;
    overflow-x: hidden;
    &.mnemonic{
        word-spacing: 15px;
        text-align: center;
    }
}
.copy-field::-webkit-scrollbar
{
	width: 9px;
	background-color: $neutral-600;
}

.copy-field::-webkit-scrollbar-thumb
{
	border-radius: 4px;
	-webkit-box-shadow: inset 0 0 6px rgba(0,0,0,.3);
    background-color: $neutral-500;
    z-index: -9999999999999999999999999999999;
}

.copy-button{
    position: absolute;
    right: 0;
    bottom: 0px;
    z-index: 10;
}
