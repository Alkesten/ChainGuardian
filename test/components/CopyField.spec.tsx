import * as React from "react";
import * as renderer from "react-test-renderer";
import {CopyField, MnemonicCopyField} from "../../src/renderer/components/CopyField/CopyField";

describe("CopyField", () => {
    it("renders correctly", () => {
        const tree = renderer
            .create(<CopyField value="Test text for copy"/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe("MnemonicCopyField", () => {
    it("renders correctly", () => {
        const tree = renderer
            .create(<MnemonicCopyField value="mnemonic text for copy"/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});