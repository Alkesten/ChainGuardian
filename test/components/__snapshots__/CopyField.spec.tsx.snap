// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CopyField renders correctly 1`] = `
Array [
  <h3
    className="copy-field-label"
  />,
  <div
    className="copy-field "
  >
    <div
      className="copy-field-body"
      style={
        Object {
          "textAlign": undefined,
        }
      }
    >
      Test text for copy
    </div>
    <button
      className="copy-button"
      onClick={[Function]}
    >
      <div
        className="__react_component_tooltip place-right type-dark "
        data-id="tooltip"
      >
        Copy All
      </div>
      <img
        className="icon copy"
        data-place="top"
        data-tip="Copy all"
        src="test-file-stub"
      />
    </button>
  </div>,
]
`;

exports[`MnemonicCopyField renders correctly 1`] = `
Array [
  <h3
    className="copy-field-label"
  />,
  <div
    className="copy-field mnemonic"
  >
    <div
      className="copy-field-body"
      style={
        Object {
          "textAlign": undefined,
        }
      }
    >
      mnemonic text for copy
    </div>
    <button
      className="copy-button"
      onClick={[Function]}
    >
      <div
        className="__react_component_tooltip place-right type-dark "
        data-id="tooltip"
      >
        Copy All
      </div>
      <img
        className="icon copy"
        data-place="top"
        data-tip="Copy all"
        src="test-file-stub"
      />
    </button>
  </div>,
]
`;
